python/lance/__init__.py

```diff
@@ -23,26 +23,33 @@
 
 __version__ = version.__version__
 
-from lance.lib import LanceFileFormat, WriteTable, _wrap_dataset
+from lance.lib import LanceFileFormat, WriteTable, _lance_dataset_write, _wrap_dataset
 from lance.types import register_extension_types
 
 __all__ = ["dataset", "write_table", "scanner", "LanceFileFormat", "__version__"]
 
 register_extension_types()
 
 
-def dataset(uri: str, **kwargs) -> ds.FileSystemDataset:
+def dataset(uri: str, version: Optional[int] = None, **kwargs) -> ds.FileSystemDataset:
     """
     Create an Arrow Dataset from the given lance uri.
 
     Parameters
     ----------
     uri: str
         The uri to the lance data
+    version: optional, int
+        If specified, load a specific version of the dataset.
     """
-    fmt = LanceFileFormat()
-    dataset = ds.dataset(uri, format=fmt, **kwargs)
-    return _wrap_dataset(dataset)
+    # TODO: check the layout first.
+    if version:
+        pass
+    else:
+        # Backward compatible to lance format w/o versioning support.
+        fmt = LanceFileFormat()
+        dataset = ds.dataset(uri, format=fmt, **kwargs)
+        return _wrap_dataset(dataset)
 
 
 def write_table(table: pa.Table, destination: Union[str, Path], batch_size: int = 1024):
```

Comment:
is there a reason to keep both `write_table` and `write_dataset`?

Reply:
Do we still use `write_table` to write "old plain version" of dataset? If no, let's delete this function then.